cdap-storage-spi/src/test/java/io/cdap/cdap/spi/data/StructuredTableTest.java

```diff
@@ -175,6 +199,62 @@ public void testSimpleScan() throws Exception {
     // TODO: test end only range
   }
 
+  @Test
+  public void testMultiScan() throws Exception {
+    int max = 100;
+
+    // Write multiple rows with sequential keys
+    writeSimpleStructuredRows(max, "");
+
+    // Empty range scan
+    List<Collection<Field<?>>> result = runMultiScan(Collections.emptyList(), Integer.MAX_VALUE, KEY, KEY2);
+    Assert.assertTrue(result.isEmpty());
+
+    // Multiscan with a range that scans all.
+    Collection<Range> ranges = Arrays.asList(
+      Range.create(Collections.singleton(Fields.intField(KEY, 0)), Range.Bound.INCLUSIVE,
+                   Collections.singleton(Fields.intField(KEY, 10)), Range.Bound.EXCLUSIVE),
+      Range.all());
+    result = runMultiScan(ranges, Integer.MAX_VALUE, KEY, KEY2);
+    List<Collection<Field<?>>> expected = new ArrayList<>();
+    for (int i = 0; i < max; i++) {
+      expected.add(Arrays.asList(Fields.intField(KEY, i), Fields.longField(KEY2, (long) i)));
+    }
+    Assert.assertEquals(expected, result);
+
+    // Constructs multiple scan ranges.
+    ranges = new ArrayList<>();
+    expected = new ArrayList<>();
+    for (int i = 0; i < 5; i++) {
+      int begin = i * 20;
+      int end = begin + 10;
+      ranges.add(Range.create(Collections.singleton(Fields.intField(KEY, begin)), Range.Bound.INCLUSIVE,
+                              Collections.singleton(Fields.intField(KEY, end)), Range.Bound.EXCLUSIVE));
+      for (int j = begin; j < end; j++) {
+        expected.add(Arrays.asList(Fields.intField(KEY, j), Fields.longField(KEY2, (long) j)));
+      }
+    }
+
+    // Scan without limit
+    result = runMultiScan(ranges, Integer.MAX_VALUE, KEY, KEY2);
+    Assert.assertEquals(expected, result);
+
+    // Scan with limit
+    int limit = 25;
+    result = runMultiScan(ranges, limit, KEY, KEY2);
+    Assert.assertEquals(expected.subList(0, limit), result);
+  }
+
+  private List<Collection<Field<?>>> runMultiScan(Collection<Range> ranges, int limit,
+                                                  String...outputFields) throws Exception {
```

Comment:
should have space after `String...`
https://docs.oracle.com/javase/tutorial/java/javaOO/arguments.html#varargs